src/firefly/java/edu/caltech/ipac/firefly/server/query/ObsCoreMetadataQuery.java

```diff
@@ -0,0 +1,107 @@
+package edu.caltech.ipac.firefly.server.query;
+
+import edu.caltech.ipac.firefly.data.TableServerRequest;
+import edu.caltech.ipac.firefly.server.util.Logger;
+import edu.caltech.ipac.firefly.server.util.QueryUtil;
+import edu.caltech.ipac.table.DataGroup;
+import edu.caltech.ipac.table.DataObject;
+import edu.caltech.ipac.table.DataType;
+import edu.caltech.ipac.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.stream.Collectors;
+
+import static edu.caltech.ipac.firefly.server.query.ObsCoreMetadataQuery.*;
+import static edu.caltech.ipac.firefly.server.query.UwsJobProcessor.getTableResult;
+
+
+@SearchProcessorImpl(id = ID, params = {
+        @ParamDoc(name = SVC_URL, desc = "base TAP url endpoint excluding '/sync'"),
+        @ParamDoc(name = OBSCORE_TNAME, desc = "obscore schema.table"),
+        @ParamDoc(name = COLUMNS, desc = "columns to retrieve metadata for"),
+})
+public class ObsCoreMetadataQuery extends EmbeddedDbProcessor {
+    public static final String ID = "ObsCoreMetadataQuery";
+    public static final String SVC_URL = "serviceUrl";
+    public static final String OBSCORE_TNAME = "obscoreTable";
+    public static final String COLUMNS = "columns";
+    public static final String QUERY_FRAGMENT = "/sync?REQUEST=doQuery&LANG=ADQL&QUERY=";
+    public static final Logger.LoggerImpl LOGGER = Logger.getLogger();
+
+
+    public DataGroup fetchDataGroup(TableServerRequest request) throws DataAccessException {
+        List<String> columns = StringUtils.asList(request.getParam(COLUMNS), ",");
+        List<Future<DataGroup>> futures = new ArrayList<>();
+        List<Throwable> exceptions = new ArrayList<>();
+
+        // Submit queries for each column in parallel
+        ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
You don't want to use edu.caltech.ipac.firefly.server.ServerContext#SHORT_TASK_EXEC?  
It should work the same.
But, if you want to create your own, wrap `executor.shutdown();` around a try/catch/finalize` to ensure it is cleaned up properly.

Reply:
`SHORT_TASK_EXEC` was the reason I was getting null pointer exceptions I shared with you earlier. That's why I created my own executor.

About cleanup are you referring to something like this: https://github.com/Caltech-IPAC/firefly/blob/a6b372576b1558ec51b7e305fc2b864369f773bf/src/firefly/java/edu/caltech/ipac/firefly/server/ServerContext.java#L870-L877 ?

If yes, I don't know what to put in finally.

Reply:
```java
ExecutorService executor = Executors.newCachedThreadPool();
try {
  // your code
} finally {
    executor.shutdown(); 
}
```
This ensure shutdown is called when encountering uncaught exceptions.